source/Parser.cpp

```diff
@@ -1837,6 +1842,27 @@ PackageImportItemSyntax* Parser::parsePackageImportItem() {
     return alloc.emplace<PackageImportItemSyntax>(package, doubleColon, item);
 }
 
+DPIImportExportSyntax* Parser::parseDPIImportExport(ArrayRef<AttributeInstanceSyntax*> attributes) {
+    auto keyword = consume();
+    auto stringLiteral = expect(TokenKind::StringLiteral);
+    if (stringLiteral.valueText() != "DPI-C" && stringLiteral.valueText() != "DPI") {
+        addError(DiagCode::ExpectedDPISpecString, stringLiteral.location());
+    }
+    Token property, name, equals;
+    if (keyword.kind == TokenKind::ImportKeyword && (peek(TokenKind::ContextKeyword) || peek(TokenKind::PureKeyword))) {
+        property = consume();
+    }
+    if (peek(TokenKind::Identifier)) {
+        name = consume();
+        equals = expect(TokenKind::Equals);
+    }
+    if (property.kind == TokenKind::PureKeyword && !peek(TokenKind::FunctionKeyword)) {
+        addError(DiagCode::ExpectedFunctionKeyword, peek().location());
```

Comment:
I don't think we should have a separate error for this. The expect() function can already build a good error for missing keywords, we should try to use that in some way. Also if there isn't a function or task here, parseFunctionPrototype() will just take whatever token happens to be there and pretend it's the keyword, which seems bad.